doc/10-Channels.md

```diff
@@ -1,88 +1,355 @@
+
+Its `result` is expected to be a JSON object with the `json` fields defined in the
+[`Info` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#Info).
+The `config_attrs` field must be an array of JSON objects according to the
+[`ConfigOption` type](https://pkg.go.dev/github.com/icinga/icinga-notifications/pkg/plugin#ConfigOption).
+Those attributes define configuration options for the channel to be set via the `SetConfig` method.
+Furthermore, they are used for channel configuration in Icinga Notifications Web.
+
+##### Example GetInfo Request
+
 ```json
 {
-  "method": "Add",
-  "params": {
-    "num1": 5,
-    "num2": 3
+  "method": "GetInfo",
+  "id": 1
+}
+```
+
+##### Example GetInfo Response
+
+```json
+{
+  "result": {
+    "name": "Minified Webhook",
+    "version": "0.0.0-gf369a11-dirty",
+    "author": "Icinga GmbH",
+    "config_attrs": [
+      {
+        "name": "url_template",
+        "type": "string",
+        "label": {
+          "de_DE": "URL-Template",
+          "en_US": "URL Template"
+        },
+        "help": {
+          "de_DE": "URL, optional als Go-Template über das zu verarbeitende plugin.NotificationRequest.",
+          "en_US": "URL, optionally as a Go template over the current plugin.NotificationRequest."
+        },
+        "required": true,
+        "min": null,
+        "max": null
+      },
+      {
+        "name": "response_status_codes",
+        "type": "string",
+        "label": {
+          "de_DE": "Antwort-Status-Codes",
+          "en_US": "Response Status Codes"
+        },
+        "help": {
+          "de_DE": "Kommaseparierte Liste erwarteter Status-Code der HTTP-Antwort, z.B.: 200,201,202,208,418",
+          "en_US": "Comma separated list of expected HTTP response status code, e.g., 200,201,202,208,418"
+        },
+        "default": "200",
+        "min": null,
+        "max": null
+      }
+    ]
   },
-  "id": 2020
+  "id": 1
+}
+```
+
+#### SetConfig
+
+The `SetConfig` method configures the channel.
+
+The Icinga Notifications daemon is going to call this method at least once on each channel
+before sending the first notifications to initialize the channel plugin.
+
+The passed JSON object in the request's `param` field reflects the objects from `GetInfo`'s `config_attrs`.
+Each object within the `config_attrs` array must be configurable by using its `name` attribute as the key together with
+the desired configuration value, being of the specified type in the `type` field.
+
+To illustrate, the URL template from the above output is configurable with the following JSON object passed in `params`:
+
+```json
+{
+  "url_template": "http://localhost:8000/update/{{.Incident.Id}}"
 }
 ```
 
+If the channel plugin successfully configured itself, a response without an `result` must be returned.
+Otherwise, if the channel decides that the provided configuration is faulty, an `error` must be returned.
+This may happen if, i.e., an invalid configuration value was given.
```

Comment:
I think you don't want to use [`i.e.`](https://dictionary.cambridge.org/dictionary/english/ie) here, but `for instance/example` instead.

Reply:
There will be a day when I finally learn how to correctly use _e.g._ and _i.e._